packages/svelte/src/internal/client/dev/tracing.js

```diff
@@ -0,0 +1,180 @@
+/** @import { Derived, Reaction, Signal, Value } from '#client' */
+import { UNINITIALIZED } from '../../../constants.js';
+import { snapshot } from '../../shared/clone.js';
+import { define_property } from '../../shared/utils.js';
+import { DERIVED, STATE_SYMBOL } from '../constants.js';
+import { active_reaction, captured_signals, set_captured_signals, untrack } from '../runtime.js';
+
+/** @type { any } */
+export let tracing_expressions = null;
+
+/**
+ * @param { Value } signal
+ * @param { { read: Error[] } } [entry]
+ */
+function log_entry(signal, entry) {
+	const debug = signal.debug;
+	const value = signal.v;
+
+	if (value === UNINITIALIZED) {
+		return;
+	}
+
+	if (debug) {
+		var previous_captured_signals = captured_signals;
+		var captured = new Set();
+		set_captured_signals(captured);
+		try {
+			untrack(() => {
+				debug();
+			});
+		} finally {
+			set_captured_signals(previous_captured_signals);
+		}
+		if (captured.size > 0) {
+			for (const dep of captured) {
+				log_entry(dep);
+			}
+			return;
+		}
+	}
+	const type = (signal.f & DERIVED) !== 0 ? 'derived' : 'state';
+	const current_reaction = /** @type {Reaction} */ (active_reaction);
+	const status =
+		signal.version > current_reaction.version || current_reaction.version === 0 ? 'dirty' : 'clean';
+
+	// eslint-disable-next-line no-console
+	console.groupCollapsed(
+		`%c${type}`,
+		status !== 'clean'
+			? 'color: CornflowerBlue; font-weight: bold'
+			: 'color: grey; font-weight: bold',
+		typeof value === 'object' && STATE_SYMBOL in value ? snapshot(value, true) : value
+	);
+
+	if (type === 'derived') {
```

Comment:
goes with https://github.com/sveltejs/svelte/pull/14290/files#r1852252100

```suggestion
	if (type === '$derived') {
```